In [ ]:
import pandas as pd

In [ ]:
from google.colab import files

uploaded = files.upload()

Saving Advanced_pandas_sales_dataset.csv to Advanced_pandas_sales_dataset.csv


In [ ]:
import os

os.listdir()

['.config', 'Advanced_pandas_sales_dataset.csv', 'sample_data']

In [ ]:
df = pd.read_csv("Advanced_pandas_sales_dataset.csv")

Q1 Dataset Exploration
Perform the following exploratory steps:

*   Display the first 10 rows
*   Display the dataset structure using info()
*   Find the number of rows and columns
*   Identify the data types of each column



In [ ]:
# Display the first 10 rows

print(df.iloc[0:10])

# Display the dataset structure using info()

print(df.info())

# Find the number of rows and columns

print(df.shape)

# Identify the data types of each column

print(df.dtypes)

   Order_ID  Order_Date Customer       City Region   Product     Category  \
0      1001  2024-10-01    Priya  Hyderabad   East     Mouse  Accessories   
1      1002  2024-05-03    Sneha    Kolkata   East   Monitor  Electronics   
2      1003  2024-01-26    Anita     Mumbai   West     Phone  Electronics   
3      1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
4      1005  2024-10-15    Rahul  Bangalore  South   Monitor  Electronics   
5      1006  2024-07-04    Pooja  Bangalore   West   Printer       Office   
6      1007  2024-06-24      Dev    Chennai   East   Monitor  Electronics   
7      1008  2024-03-18    Kavya    Kolkata   East     Phone  Electronics   
8      1009  2024-10-31    Kavya  Ahmedabad   East     Phone  Electronics   
9      1010  2024-12-25    Pooja  Ahmedabad  North     Phone  Electronics   

   Sales  Quantity  Discount  Profit  
0   1735         7      0.00  307.52  
1    807         2      0.15   49.89  
2   2328         1      0.20   66.4

Q2 Sales Filtering Analysis
Filter the dataset to display:
Explain what insights you observe

1.   Orders where Sales > 3500
2.   Orders where Region = "East" and Sales > 2500



In [ ]:
# Orders where Sales > 3500

print(df[df["Sales"] > 3500])

# Orders where Region = "East" and Sales > 2500

print(df[(df["Region"]=="East") & (df["Sales"]>2500)])


     Order_ID  Order_Date Customer       City Region   Product     Category  \
3        1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
4        1005  2024-10-15    Rahul  Bangalore  South   Monitor  Electronics   
7        1008  2024-03-18    Kavya    Kolkata   East     Phone  Electronics   
8        1009  2024-10-31    Kavya  Ahmedabad   East     Phone  Electronics   
10       1011  2024-06-30     Neha      Delhi  South   Printer       Office   
..        ...         ...      ...        ...    ...       ...          ...   
219      1220  2024-03-22    Meera  Bangalore   West     Phone  Electronics   
222      1223  2024-03-04    Meera  Ahmedabad  South     Phone  Electronics   
223      1224  2024-01-07      Dev  Bangalore   East    Tablet  Electronics   
225      1226  2024-06-14    Anita  Ahmedabad   East    Laptop  Electronics   
229      1230  2024-04-13    Meera    Kolkata  North   Monitor  Electronics   

     Sales  Quantity  Discount   Profit  
3     393

**Business Insight**

High-value sales are concentrated in selected regions. East region has several orders above ₹2500, indicating strong customer demand and better revenue generation.

Q3 Customer Purchase Analysis

Identify the top 5 customers based on total sales.

Hint: Use GroupBy with aggregation.

In [ ]:
top5 = df.groupby("Customer")["Sales"].sum().sort_values(ascending=False).head(5)

print(top5)

Customer
Nitin    59820
Raj      58416
Rahul    52291
Dev      48500
Isha     46818
Name: Sales, dtype: int64


Q4 Regional Sales Performance

Calculate the total sales generated by each region.

Then identify:
*   Which region generated the highest revenue?




In [ ]:
region_sales = df.groupby("Region")["Sales"].sum()

print(region_sales)

print("Highest Revenue Region:",region_sales.idxmax())


Region
East     232675
North    158210
South    183108
West     170227
Name: Sales, dtype: int64
Highest Revenue Region: East


Q5 Product Performance Analysis

*   Calculate the total quantity sold for each product.
*   Which product appears to be the most popular?



In [ ]:
# Which product appears to be the most popular?

product_qty = df.groupby("Product")["Quantity"].sum()
print(product_qty)

#Which product appears to be the most popular?

print("Most Popular Product:",product_qty.idxmax())


Product
Headphones    115
Keyboard      110
Laptop        140
Monitor       156
Mouse         122
Phone         151
Printer       108
Tablet        150
Name: Quantity, dtype: int64
Most Popular Product: Monitor


Q6 Multi-Level Grouping

*   Group the dataset by: Region and Category
*   Calculate the total sales for each combination.
*   What insights can you derive from this analysis?




In [ ]:
group_data = df.groupby(["Region","Category"])["Sales"].sum()

print(group_data)

Region  Category   
East    Accessories     57646
        Electronics    160907
        Office          14122
North   Accessories     62052
        Electronics     70023
        Office          26135
South   Accessories     80914
        Electronics     86690
        Office          15504
West    Accessories     64811
        Electronics     84128
        Office          21288
Name: Sales, dtype: int64


**Business Insight**

Multi-level grouping helps compare category-wise sales across different regions. It enables businesses to identify which product categories perform best in each region and supports region-specific marketing and inventory decisions.


Q7 Pivot Table Creation
Create a pivot table showing:

*   Rows → Region
*   Columns → Product
*   Values → Total Sales

Which region sells the most laptops?



In [ ]:
pivot = pd.pivot_table(
    df,
    index="Region",
    columns="Product",
    values="Sales",
    aggfunc="sum"
)

print(pivot)

Product  Headphones  Keyboard  Laptop  Monitor  Mouse  Phone  Printer  Tablet
Region                                                                       
East          23899     10845   55568    31299  22902  39667    14122   34373
North         15450     20538    4540    14922  26064  30535    26135   20026
South         24111     23480   23029    19167  33323  18218    15504   26276
West          17159     38499   10826    18178   9153  27850    21288   27274


From the pivot table

The sales of Laptops in each region are:

East - 55568

North - 4540

South - 23029

West - 10826

The East region has most sales of laptop.

Q8 Profitability Analysis

*   Calculate the average profit for each product category.
*   Which category is most profitable?



In [ ]:
avg_profit = df.groupby("Category")["Profit"].mean()

print(avg_profit)

print("Most Profitable Category:",avg_profit.idxmax())


Category
Accessories    410.603086
Electronics    396.462114
Office         471.938462
Name: Profit, dtype: float64
Most Profitable Category: Office


Q9 Discount Impact Study

Analyze the relationship between discount and profit.

Tasks:

*   Calculate the average profit for each discount level
*   Identify whether higher discounts lead to lower profit




In [ ]:
discount_profit = df.groupby("Discount")["Profit"].mean()

print(discount_profit)

Discount
0.00    751.210588
0.05    531.368039
0.10    346.460513
0.15    237.191400
0.20     90.026923
Name: Profit, dtype: float64


**Business Insight**

As the discount percentage increases, the average profit decreases. This indicates that offering higher discounts reduces profitability, so businesses should apply discounts carefully to maintain healthy profit margins.

Q10 End-to-End Data Wrangling Task

Perform the following workflow:

1.   Filter the dataset where Sales > 3000
2.   Group data by Region and Product
3.   Calculate total sales
4.   Reshape the result into a pivot table

Explain the business insights obtained from this analysis.

In [ ]:
# Step 1: Filter the data
filtered = df[df["Sales"] > 3000]

# Step 2: Group by Region and Product and calculate Total Sales
grouped = filtered.groupby(["Region", "Product"])["Sales"].sum().reset_index()

# Display Total Sales
print("Total Sales by Region and Product:")
print(grouped)

# Step 3: Convert into Pivot Table
pivot_table = grouped.pivot(
    index="Region",
    columns="Product",
    values="Sales"
)

# Display Pivot Table
print("\nPivot Table:")
print(pivot_table)

Total Sales by Region and Product:
   Region     Product  Sales
0    East  Headphones  14469
1    East    Keyboard   9642
2    East      Laptop  44096
3    East     Monitor  16420
4    East       Mouse  19000
5    East       Phone  29500
6    East     Printer   3696
7    East      Tablet  29229
8   North  Headphones   5326
9   North    Keyboard  15653
10  North      Laptop   4540
11  North     Monitor   5087
12  North       Mouse  22209
13  North       Phone  24670
14  North     Printer  17185
15  North      Tablet   3392
16  South  Headphones  10026
17  South    Keyboard  19249
18  South      Laptop  17950
19  South     Monitor  14921
20  South       Mouse  26928
21  South       Phone  18218
22  South     Printer  14143
23  South      Tablet  19613
24   West  Headphones  13898
25   West    Keyboard  33380
26   West      Laptop   5922
27   West     Monitor  14354
28   West       Mouse   3511
29   West       Phone  15653
30   West     Printer  12648
31   West      Tablet  17185

Pivot T

**Business Insights**

*   The analysis includes only high-value orders (Sales > 3000).
*   The pivot table shows the total sales of each product across different regions.
*   East contributes significantly to high-value sales, especially for Laptops.
*   The results help identify the best-performing products in each region and support decisions related to inventory planning, sales strategy, and regional marketing.


